tests/all/boxed.rs

```diff
@@ -12,3 +12,13 @@ fn into_raw_aliasing() {
     let mut_ref = unsafe { &mut *raw };
     dbg!(mut_ref);
 }
+
+#[cfg(feature = "serde")]
+#[test]
+fn test_box_serializes() {
+    let bump = Bump::new();
+    let boxed = Box::new_in(1, &bump);
+    assert_eq!(serde_json::to_string(&boxed).unwrap(), "1");
+    let boxedStr = Box::new_in("a", &bump);
+    assert_eq!(serde_json::to_string(&boxedStr).unwrap(), "\"a\"");
```

Comment:
Can we also test a few other things:

* tuple
* struct w/ a couple fields
* enum w/ a couple variants
* vec w/ a couple elements
* hash map
* hash set

It should be fine to add a dev dependency on `serde_derive` if it makes this easier, fwiw.

Also, no need to explicitly write out the expected JSON; should be able to call `serde_json::to_string(&unboxed_value)` and then `serde_json::to_string(&boxed_value)` and compare the results for equality, I believe. Or compare `std::boxed::Box<T>` with `bumpalo::boxed::Box<'_, T>`. Might be worth writing a little helper that does this dance, rather than writing it out in full for all the cases above.

Reply:
I've added a struct with an enum and option. Hopefully that's sufficient, or would you like me to add more such as the hash map/set?